portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -154,4 +170,35 @@ private int getNumQuoteDisplayed(HttpServletRequest request, int defaultNumToDis
       return defaultNumToDisplay; 
     }
   }
+
+  /**
+   * Extracts the attributes of a quoteEntity and puts them into a new
+   * instance of Quote. 
+   */ 
+  private Quote extractQuoteFromEntity(Entity quoteEntity) {
+    long quoteId = quoteEntity.getKey().getId();
+    String text = (String) quoteEntity.getProperty(TEXT);
+    long timestampMillis = (long) quoteEntity.getProperty(TIMESTAMP);
+    String userEmail = (String) quoteEntity.getProperty(USER_EMAIL); 
+    String nickname = (String) quoteEntity.getProperty(NICKNAME); 
+    Quote newQuote = new Quote(quoteId, text, timestampMillis, userEmail, nickname);
+    return newQuote; 
+  } 
+
+  /**
+   * Returns the nickname of the user with id, or null if the user has not set a nickname.
```

Comment:
Consider clarifying in the method comment that it can return null.

Reply:
I incorporated Optional in this method.